### PR TITLE
Added Travis CLI for build Automation ( Ready for merge )

### PR DESCRIPTION
**Pull request by niroshn**
#71 
Successfully added Travis build file ( .travis.yml )
It is working fine.

After every PR travis automatically run these build.
Then Developer can catch their conflict and error.

### CODE REVIEW

**Comment by niroshn** [comment]
When Running Build tassk
travis-build

**Comment by niroshn** [comment]
After successful build 

travis

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+sudo: false
+
+language: node_js
+
+node_js:
+  - "4"
```
Comment (obsidianart) [inline .travis.yml:6]: why 4?

**Comment by obsidianart** [comment]
@Treora is "make" the right script? can we change it now to npm run make if that's the case so the automation can rely on npm scripts?

**Comment by Treora** [comment]
> @Treora is "make" the right script?

My tradition is to put a Makefile in every single project, so that users don't have to be bothered which command to run (like a [universal install script](https://www.xkcd.com/1654/)). Also it serves as a reference of the possible actions one may want to do.

For this configuration file, calling the `npm` or `gulp` commands directly would be fine too. Calling `make` does however have the advantage that the Makefile is also tested.

File: .travis.yml (hunk 1 above)
Comment (niroshn) [reply .travis.yml:6]: This is version of node.js

**Comment by niroshn** [comment]
@Treora Is there any other issue in this PR

**Comment by Treora** [comment]
> @Treora Is there any other issue in this PR

#71

**Comment by Treora** [comment]
Oh you mean literal issues. :)

I don't know anything about Travis config files, so I cannot judge. We had not really dediced whether to use Travis, or another platform, so that is the main question still.

**Comment by niroshn** [comment]
yes literal issues.

Travis is a most popular CI platform. 
After trying Travis CLI we can remove it If you need.

**Comment by niroshn** [comment]
I added eslint in here #72

**Comment by danrl** [comment]
It's not doing any harm using TravisCI for a while. We can switch whenever we like.
I support the move to use TravisCI and evaluate after a while if it was helpful and improved code quality or not.

File: .travis.yml (hunk 1 above)
Comment (obsidianart) [reply .travis.yml:6]: yes, why 4 and not 7?

**Comment by obsidianart** [comment]
@Treora I would say ```npm run``` and ```npm build``` are more standard in js projects than make

**Comment by Treora** [comment]
It would indeed not be a problem to start with Travis. I just thought now would be a good moment to look at the options, though I have not deemed CI important enough to investigate myself.

As for `make` versus `npm run` and such, `npm` is indeed more standard in javascript projects nowadays; however there is more than javascript, and more than nowodays. Anyway, I have no preference for whether this script runs `make` or `npm run`.

I would remove the eslint from this PR, that's being worked on in @reficul31's [branch](https://github.com/reficul31/webmemex-extension/tree/issue_72). Also is there a reason for the `npm-debug.log` to be there?

**Comment by niroshn** [comment]
@Treora  Shall we try with Travis CL then we can decide continue or not

File: .travis.yml (hunk 1 above)
Comment (niroshn) [reply .travis.yml:6]: Yes We can use 7 . 
 4 mean latest 4.x release
7 mean latest 7.x release
Will update PR for 7

**Comment by bwbroersma** [comment]
@Treora also look into [drone.io](https://github.com/drone/drone), it's very easy to self host.

**Comment by Treora** [comment]
@bwbroersma: thanks for the Drone tip. It looks like it requires a couple of extra steps though, so I opted for trying Travis for now, as I am not even convinced yet of the added value of these build&test services.

@niroshannrsh: as eslint has been added in #83, I just looked at your first commit in this PR. I only saw the need for two of the lines in your `travis.yml` config, so I started from scratch instead. I wonder what was the reason to add the config for `addons.apt`, and for the `install` commands?

Issue solved in f317827, closing this PR.

**Pull request closed without merging**